compiler/il/OMRBlock.cpp

```diff
@@ -1336,21 +1336,13 @@ OMR::Block::splitEdge(TR::Block *from, TR::Block *to, TR::Compilation *c, TR::Tr
 
       if (isExceptionEdge)
          {
-         if (to->isOSRCatchBlock())
-            newBlock->setHandlerInfoWithOutBCInfo(
-                  TR::Block::CanCatchOSR, 
-                  to->getInlineDepth(), 
-                  -1, 
-                  to->getOwningMethod(), 
-                  c);
-         else
-            newBlock->setHandlerInfo(
-               to->getCatchType(),
-               to->getInlineDepth(),
-               to->getHandlerIndex(),
-               to->getOwningMethod(),
-               c);
-            TR::ResolvedMethodSymbol *method = c->getMethodSymbol();
+         newBlock->setHandlerInfo(
+            to->getCatchType(),
+            to->getInlineDepth(),
+            to->getHandlerIndex(),
+            to->getOwningMethod(),
+            c);
+         TR::ResolvedMethodSymbol *method = c->getMethodSymbol();
```

Comment:
This code should be guarded to make sure we only do this for exception edges going to non-OSR catch blocks. We should also do something nasty like return NULL if we fail to split so that it causes the code to fall over aggressively in the unsupported case.